log.go

```diff
@@ -76,33 +77,60 @@ type LeveledLogger struct {
 	stdoutOverride io.Writer
 }
 
+var apiKeyRegex *regexp.Regexp
+
+func init() {
+	var err error
+	apiKeyRegex, err = regexp.Compile("(sk_live|sk_test|rk_live|rk_test)_[a-zA-Z0-9]{2}[a-zA-Z0-9]{1,}")
```

Comment:
Lurking in: Historical API keys did not have `sk_` prefix or `live_` in them so that would not catch all of them.

I'm also curious why we're adding this. The internal ask I'm aware of is specifically about `client_secret` which is a property on PaymentIntent and SetupIntent, is this related or orthogonal?